### PR TITLE
support parse select expr as `alias_name`

### LINKED ISSUE #4840: Feature: Support select expr as `alias name`
**Summary**

Link to this issue #[26](https://github.com/datafuse-extras/sqlparser-rs/issues/26)

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

After this pr,  this SQL will return err.

```SQL
SELECT TRUE as `_`  FROM `t1` WHERE 1 <> 1 LIMIT 0;

ERROR 1105 (HY000): Code: 1005, displayText = sql parser error: Expected an identifier after AS, found: `_`.
```
The err occurs in sqlparser, the err has already been fixed [there](https://github.com/datafuse-extras/sqlparser-rs/pull/27).

## Changelog

- New Feature

## Related Issues

Fixes #4840

### CODE REVIEW

File: common/ast/src/udfs/udf_transformer.rs [hunk 1]
```diff
@@ -127,9 +127,12 @@ impl UDFTransformer {
                     op: op.clone(),
                     expr: Box::new(Self::clone_expr_with_replacement(&**expr, replacement_fn)?),
                 }),
-                Expr::Cast { expr, data_type } => Ok(Expr::Cast {
+                Expr::Cast {
+                    expr, data_type, ..
+                } => Ok(Expr::Cast {
```
Comment (TCeason) [inline common/ast/src/udfs/udf_transformer.rs:132]: @b41sh  Hi the pg_style can be set as false?

**Comment by BohuTANG** [comment]
This need a stateless test, it is easy to find the bug when the new sqlparser is introduced someday.

**Comment by TCeason** [comment]
> This need a stateless test, it is easy to find the bug when the new sqlparser is introduced someday.

Yep, you are right. Added.

File: common/ast/src/parser/expr/expr_visitor.rs [hunk 2]
```diff
@@ -57,7 +57,11 @@ pub trait ExprVisitor: Sized + Send {
             Expr::Subquery(subquery) => self.visit_subquery(subquery),
             Expr::Function(function) => self.visit_function(function).await,
             Expr::TryCast { expr, data_type } => self.visit_try_cast(expr, data_type).await,
-            Expr::Cast { expr, data_type } => self.visit_cast(expr, data_type).await,
+            Expr::Cast {
+                expr,
+                data_type,
+                pg_style: _pg_style,
```
Comment (BohuTANG) [inline common/ast/src/parser/expr/expr_visitor.rs:63]: Does the new version of sqlparser introduced `pg_style` params?

File: common/ast/src/udfs/udf_transformer.rs (hunk 1 above)
Comment (BohuTANG) [reply common/ast/src/udfs/udf_transformer.rs:132]: We don't used `pg_style`, i think it's OK here.

File: common/ast/src/parser/expr/expr_visitor.rs (hunk 2 above)
Comment (TCeason) [reply common/ast/src/parser/expr/expr_visitor.rs:63]: https://github.com/datafuse-extras/sqlparser-rs/commit/b0571e8f91cac092fc8f8873e8babcc05fb91974

Yes in this commit.

Comment (BohuTANG) [reply common/ast/src/parser/expr/expr_visitor.rs:63]: O, @b41sh should know it better :D

File: common/ast/src/udfs/udf_transformer.rs (hunk 1 above)
Comment (b41sh) [reply common/ast/src/udfs/udf_transformer.rs:132]: we can set `pg_style` as false, it is used to distinguish the field names of two different casts.
```sql
select cast(a as int8) from test;
select a::int8 from test;
```
I am working on this modification, you can set it as `false` first.

**Pull request merged**